### ISSUE TITLE
openssl not in the build order when compiling cryptography

### ISSUE THREAD

**Issue by AndreMiras** [kivy/python-for-android#1764]
### Versions

* Python: 3
* OS: Ubuntu 18.04
* Kivy: N/A
* Cython: 0.28.6

### Description

From the "official" Docker container, when trying to build `cryptography` it seems like `openssl` is not getting added in the build order when doing the dependency resolution.
The command used was:
```
python setup_testapp_python3.py apk --sdk-dir $ANDROID_SDK_HOME --ndk-dir $ANDROID_NDK_HOME --bootstrap sdl2 --requirements python3,cryptography
```
And the beginning of the log was:
```
packages are []
running apk
WARNING: You did not supply an Android package identifier, trying org.test.testapp_python3_googlendk instead.
         This may fail if this is not a valid identifier
Including testapp/main.py
Including testapp/colours.png
[WARNING]: --ndk-version is deprecated and no longer necessary, the value you passed is ignored
[INFO]:    Will compile for the following archs: armeabi-v7a
[INFO]:    Getting Android API version from user argument: 27
[INFO]:    Available Android APIs are (19, 27)
[INFO]:    Requested API target 27 is available, continuing.
[INFO]:    Getting NDK dir from from user argument
[INFO]:    Found NDK revision 17.2.4988734
[INFO]:    Getting NDK API version (i.e. minimum supported API) from user argument
[INFO]:    Found virtualenv at /usr/bin/virtualenv
[INFO]:    Found the following toolchain versions: ['4.9']
[INFO]:    Picking the latest gcc toolchain, here 4.9
[INFO]:    No existing dists meet the given requirements!
[INFO]:    No dist exists that meets your requirements, so one will be built.
[INFO]:    Found a single valid recipe set: ['asn1crypto', 'hostpython3', 'libffi', 'sdl2_image', 'sdl2_mixer', 'sdl2_ttf', 'python3', 'sdl2', 'setuptools', 'six', 'enum34', 'idna', 'ipaddress', 'pycparser', 'pyjnius', 'android', 'cffi', 'cryptography']
[INFO]:    The selected bootstrap is sdl2
[INFO]:    # Creating dist with sdl2 bootstrap
[INFO]:    Dist will have name bdisttest_python3_googlendk and requirements (python3, cryptography)
[INFO]:    Dist contains the following requirements as recipes: ['hostpython3', 'libffi', 'sdl2_image', 'sdl2_mixer', 'sdl2_ttf', 'python3', 'sdl2', 'setuptools', 'six', 'enum34', 'idna', 'ipaddress', 'pycparser', 'pyjnius', 'android', 'cffi', 'cryptography']
[INFO]:    Dist will also contain modules (asn1crypto) installed from pip
[INFO]:    -> directory context /home/user/.local/share/python-for-android/build/bootstrap_builds/sdl2-python3
[INFO]:    <- directory context /home/user/testapps
[INFO]:    Recipe build order is ['hostpython3', 'libffi', 'sdl2_image', 'sdl2_mixer', 'sdl2_ttf', 'python3', 'sdl2', 'setuptools', 'six', 'enum34', 'idna', 'ipaddress', 'pycparser', 'pyjnius', 'android', 'cffi', 'cryptography']
[INFO]:    The requirements (asn1crypto) were not found as recipes, they will be installed with pip.
[INFO]:    # Downloading recipes
```

Since `openssl` is in the `depends` list (see [cryptography recipe](https://github.com/kivy/python-for-android/blob/0e27c59/pythonforandroid/recipes/cryptography/__init__.py#L8)) I'm expecting to see `openssl` being built first.

**Comment by AndreMiras** [comment]
So this actually is a feature coming from https://github.com/kivy/python-for-android/pull/1696
Now we probably need to adapt the conditional build script to build against a full setup, e.g. testapps/setup_testapp_python3_sqlite_openssl.py

**Comment by AndreMiras** [comment]
Fixed with #1761
